packages/block-library/src/image/edit.js

```diff
@@ -87,13 +88,20 @@ export function ImageEdit( {
 		width,
 		height,
 		sizeSlug,
+		isInGallery,
```

Comment:
will switch this to a `context`

Reply:
Perhaps the context name should be generalized to something that could be reused in other blocks like the Navigation block, e.g. `isInList` or `isListItem`.

Reply:
I added a context of `isList` as this seemed slightly more natural language wise, eg. if image context.isList then treat is as a list item ... but happy to revise if I am looking at it from the wrong angle.

Also, I till had to add an image attribute of isListItem still, but [setting this based in context](https://github.com/WordPress/gutenberg/pull/24039/files#diff-f5fc3d5a78f58a5274bc196ebc657c5bR226) as I couldn't see any way to access the context in the save method, and we need to know there whether to wrap the image in an <li> or not as well as in edit ... let me know if I have missed a better way fo handling this.

Reply:
I'm not entirely sure which way to go with `isList` vs `isListItem` myself. It would be good to get some more opinions on this.

The edit markup should match the save markup as much as possible, so if it's being wrapped in a `<li>` in one, the other should match.

I would think (but could be wrong) that there should already be a way to access block context from the JS save component. But if not, the attribute is a decent workaround for now until that gets implemented.

Reply:
I'm now thinking we shouldn't be handling this via block context at all. See [this comment](https://github.com/WordPress/gutenberg/pull/24039#discussion_r461260127). Of course, implementing a new InnerBlocks prop is outside the scope of this PR, so that task would have to be done in a separate PR.

Reply:
>I would think (but could be wrong) that there should already be a way to access block context from the JS save

The docs seem to indicate that is is currently only accessible via [JS in the edit method](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-context.md#using-block-context), and doing some debugging and looking through the code seems to confirm that, but let me if you come across a way of accessing context in the save method.